sky/server/common.py

```diff
@@ -251,11 +265,28 @@ def check_server_healthy(endpoint: Optional[str] = None,) -> None:
     api_server_info = get_api_server_status(endpoint)
     api_server_status = api_server_info.status
     if api_server_status == ApiServerStatus.VERSION_MISMATCH:
+        sv, cv = api_server_info.api_version, server_constants.API_VERSION
+        assert sv is not None, 'API server version is None'
+        try:
+            server_is_older = int(sv) < int(cv)
+        except ValueError:
+            # Raised when the server version is not a numeric. A safe
+            # assumption is that our client is older since we never used
+            # a non-numeric API version before.
+            server_is_older = False
```

Comment:
Isn't it possible to have a newer client? E.g., we release a non-numeric API version and forget to update this line in the update.

Reply:
Yes, will add a unit test to ensure the logic is consistent at least at client-side